tests/Collection/BuilderCollectionFunctionalTest.php

```diff
@@ -27,8 +30,10 @@ public function testAggregate(): void
                 buckets: 2,
             ),
         );
-        // Extract the list of stages for arg type restriction
-        $pipeline = iterator_to_array($pipeline);
+
+        if ($pipelineAsArray) {
+            $pipeline = iterator_to_array($pipeline);
+        }
```

Comment:
I don't disagree that this tests the underlying behavior, but it seems a bit backwards that this operates by converting the Pipeline _back_ into an array, instead of creating a list of stages and then conditionally converting _that_ to a pipeline.

The latter approach seems more straightforward.

Reply:
You're right, I hadn't thought of it like that.